### PR TITLE
Implement debouncing in search bar using rxjs

### LINKED ISSUE #520: [Refactor] Implement debouncing for input in search bar
Remove [react-debounce-input](https://www.npmjs.com/package/react-debounce-input)
https://github.com/short-d/short/pull/516#discussion_r388382656

### PR DESCRIPTION
**Pull request by kapillamba4**
Closes #520

### CODE REVIEW

File: frontend/src/component/ui/SearchBar.tsx [hunk 1]
```diff
@@ -19,8 +20,24 @@ export class SearchBar extends Component<Props, State> {
     showAutoCompleteBox: true
   };
 
+  private onSearch$: any = new Subject();
+  private subscription: any = null;
+  componentDidMount() {
```
Comment (byliuyang) [inline frontend/src/component/ui/SearchBar.tsx]:
```suggestion

  componentDidMount() {
```

File: frontend/src/component/ui/SearchBar.tsx [hunk 2]
```diff
@@ -19,8 +20,24 @@ export class SearchBar extends Component<Props, State> {
...
+    this.subscription = this.onSearch$
+      .pipe(debounceTime(300))
```
Comment (byliuyang) [inline frontend/src/component/ui/SearchBar.tsx]: Let's extract out 300 to a constant with readable name to improve readability.

**Comment by kapillamba4** [comment]
@byliuyang, I am getting "Unhandled Rejection (ReferenceError): chrome is not defined"

**Comment by byliuyang** [comment]
@kapillamba4 Please run `yarn` and try again.

**Comment by kapillamba4** [comment]
Still the same

**Comment by kapillamba4** [comment]
Screen Shot 2020-03-10 at 10 29 28

**Comment by byliuyang** [comment]
@kapillamba4 I think it because `chrome` only available inside `Chrome`.  Please install Google Chrome for testing and file a bug for @rohithbalaji123

**Pull request merged**